src/index.js

```diff
@@ -1,390 +1,603 @@
+var DEFAULT = 0
+var RECYCLED_NODE = 1
+var TEXT_NODE = 3 // Node.TEXT_NODE
 
-  while (length-- > 2) rest.push(arguments[length])
+var XLINK_NS = "http://www.w3.org/1999/xlink"
+var SVG_NS = "http://www.w3.org/2000/svg"
 
-  while (rest.length) {
-    var node = rest.pop()
-    if (node && node.pop) {
-      for (length = node.length; length--; ) {
-        rest.push(node[length])
+var EMPTY_OBJECT = {}
+var EMPTY_ARRAY = []
+
+var map = EMPTY_ARRAY.map
+var isArray = Array.isArray
+
+var merge = function(a, b) {
+  var target = {}
+
+  for (var i in a) target[i] = a[i]
+  for (var i in b) target[i] = b[i]
+
+  return target
+}
+
+var resolved = typeof Promise === "function" && Promise.resolve()
+
+var defer = !resolved
+  ? setTimeout
+  : function(cb) {
+      return resolved.then(cb)
+    }
+
+var updateProperty = function(
+  element,
+  name,
+  lastValue,
+  nextValue,
+  eventProxy,
+  isSvg
+) {
+  if (name === "key") {
+  } else if (name === "style") {
+    for (var i in merge(lastValue, nextValue)) {
+      var style = nextValue == null || nextValue[i] == null ? "" : nextValue[i]
+      if (i[0] === "-") {
+        element[name].setProperty(i, style)
+      } else {
+        element[name][i] = style
       }
-    } else if (node != null && node !== true && node !== false) {
-      children.push(node)
     }
-  }
+  } else {
+    if (name[0] === "o" && name[1] === "n") {
+      name = name.slice(2).toLowerCase()
 
-  return typeof name === "function"
-    ? name(attributes || {}, children)
-    : {
-        nodeName: name,
-        attributes: attributes || {},
-        children: children,
-        key: attributes && attributes.key
+      if (!element.events) element.events = {}
+
+      element.events[name] = nextValue
+
+      if (nextValue == null) {
+        element.removeEventListener(name, eventProxy)
+      } else if (lastValue == null) {
+        element.addEventListener(name, eventProxy)
       }
-}
+    } else {
+      var nullOrFalse = nextValue == null || nextValue === false
 
-export function app(state, actions, view, container) {
-  var map = [].map
-  var rootElement = (container && container.children[0]) || null
-  var oldNode = rootElement && recycleElement(rootElement)
-  var lifecycle = []
-  var skipRender
-  var isRecycling = true
-  var globalState = clone(state)
-  var wiredActions = wireStateToActions([], globalState, clone(actions))
-
-  scheduleRender()
-
-  return wiredActions
-
-  function recycleElement(element) {
-    return {
-      nodeName: element.nodeName.toLowerCase(),
-      attributes: {},
-      children: map.call(element.childNodes, function(element) {
-        return element.nodeType === 3 // Node.TEXT_NODE
-          ? element.nodeValue
-          : recycleElement(element)
-      })
+      if (
+        name in element &&
+        name !== "list" &&
+        name !== "draggable" &&
+        name !== "spellcheck" &&
+        name !== "translate" &&
+        !isSvg
```

Comment:
can we not use it like below instead of multiple AND conditions
```suggestion
        var array = ["list","draggable","spellcheck", "translate"]
       if(
       name in element &&  
       !array.includes(name) && 
       name && !isSvg)
```

Reply:
Embedding all the conditions in the same `if` expression should be less expensive than a method call.

Reply:
Another reason: `ES2016 Specifications included the includes() method` 😭 

We would have to use `indexOf` and as @jorgebucaran said, `if` expression will be less expensive 🙏